Ensure we do not fail when CI leap second file is expired.

mhvk opened the pull request:
We are getting failures because travis does not quite keep its distribution up to date enough for leap second tests. So, this PR changes the test such that on CI if the leap second file is expired, the test is simply skipped. But if it is not expired, it checks that it is used.

cc @bsipocz, as it is something that should make your life that little bit easier!

mhvk commented:
Now if only it *solved* the failures...

pllim commented:
I think the `ON_CI` definition is correct, so somehow `iers.LeapSeconds.open(SYSTEM_FILE).expires < self.good_enough` does not apply?

mhvk commented:
Yes, I think the test I do is not exactly the same as what is used in the real code...  Will push a correction shortly...

mhvk commented:
Turns out our built-in leap-second file was expired, which was the real underlying source of the errors. I now updated that as well, and added a test that will make it clearer if that is the real problem.

pllim commented on astropy/utils/iers/data/Leap_Second.dat:
```diff
@@ -1,10 +1,10 @@
 #
 #
-#  File expires on 28 June 2020
+#  File expires on 28 December 2020
```
Does this need a change log?

mhvk replied:
Hmm, so far we have not done that for IERS_B at least. I think it is something a user should be able to assume has happened....

pllim commented on astropy/utils/iers/tests/test_leap_second.py:
```diff
@@ -181,13 +186,17 @@ def test_erfa_found(self):
             ls = iers.LeapSeconds.open()
         assert ls.meta['data_url'] == iers.IERS_LEAP_SECOND_FILE
+        # The built-in file should be kept up to date.
+        # TODO: would be nice to have automatic PRs for this!
```
Is this request the same as the one at #1697 ?

mhvk replied:
Yes, but now every half year we'll get a reminder... (more seriously, will add this to #1697)

mhvk replied:
and I see that issue ends with me promising to open a PR to the release procedure... :man_facepalming:

bsipocz replied:
> release procedure

Not sure how I feel about it being part of the release procedure. However it certainly can be a "last before release" type of reminder, similarly to the bundled libraries. See https://github.com/astropy/astropy/issues/9018

bsipocz replied:
maybe we can make this a cron only test (or simply just flag with remote_data thus allowed to fail on PRs), so as not to introduce false positives while the update PR is not merged?

pllim replied:
:+1: to `remote_data` with a comment to explain why. But it will have to be a follow-up PR.

mhvk replied:
OK, I'll make a quick PR that marks this remote_data - agreed that we don't really want to have regular PRs start failing with it!

pllim commented:
`mpldev` failure is unrelated.

mhvk commented:
@pllim, @bsipocz - if all tests pass here, I plan to go ahead and merge so that we get rid of the CI failures (and my upcoming release doc PR will not break!).

mhvk commented:
Grr, small error still....

pllim commented:
Just to be sure, are any of the numpy-dev job failures related to this?

https://travis-ci.org/astropy/astropy/jobs/644009511

mhvk commented:
One of those is `numpy-dev`, I think (the `time` one); others seem related to `jplephem`, with one a download error and the other a precision problem not captured by `FLOAT_CMP` which I've seen before (and am now fed up with!).
